qiime/plugin/testing.py

```diff
@@ -84,3 +90,33 @@ def assertSemanticTypeRegisteredToFormat(self, semantic_type, exp_format):
             obs_format, exp_format,
             "Expected semantic type %r to be registered to format %r, not %r."
             % (semantic_type, exp_format, obs_format))
+
+    def transform_format(self, source_format, target, **kwargs):
+        # Guard any non-QIIME2 Format sources from being tested
+        self.assertTrue(issubclass(source_format, FormatBase))
+
+        transformer = self.get_transformer(source_format, target)
+
+        source_path = None
+        if 'filename' in kwargs:
+            source_path = kwargs['filename']
+        elif 'filenames' in kwargs:
+            source_path = self.temp_dir.name
+            for filename in kwargs['filenames']:
+                filepath = self.get_data_path(filename)
+                shutil.copy(filepath, source_path)
+        else:
+            self.fail("The helper method `transform_format` requires either "
+                      "a filename or a sequence of filenames supplied as their"
+                      " respective keyword arguments.")
+        input = source_format(source_path, mode='r')
+
+        obs = transformer(input)
+
+        if issubclass(target, DirectoryFormat):
```

Comment:
We should probably test for `FormatBase` here, instead of `DirectoryFormat`, since we can be transforming to to `FileFormat`, which has the same `pathlib.Path`, `str`, etc properties as `DirectoryFormat`